pkg/fanal/image/registry/azure/azure.go

```diff
@@ -24,13 +35,26 @@ func (r *Registry) CheckOptions(domain string, _ types.RegistryOptions) error {
 }
 
 func (r *Registry) GetCredential(ctx context.Context) (string, string, error) {
-	credStore, err := NewACRCredStore()
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return "", "", xerrors.Errorf("unable to generate acr credential error: %w", err)
+	}
+	aadToken, err := cred.GetToken(ctx, policy.TokenRequestOptions{Scopes: []string{scope}})
 	if err != nil {
-		return "", "", xerrors.Errorf("ACR credential error: %w", err)
+		return "", "", xerrors.Errorf("unable to get an access token: %w", err)
 	}
-	token, err := credStore.Get(ctx, r.domain)
+	rt, err := refreshToken(ctx, aadToken.Token, r.domain)
 	if err != nil {
-		return "", "", xerrors.Errorf("unable to get a token: %w", err)
+		return "", "", xerrors.Errorf("unable to refresh token: %w", err)
+	}
+	return "00000000-0000-0000-0000-000000000000", *rt.RefreshToken, err
+}
+
+func refreshToken(ctx context.Context, accessToken string, domain string) (containerregistry.RefreshToken, error) {
+	tenantID := os.Getenv("AZURE_TENANT_ID")
```

Comment:
Does it mean only `AZURE_TENANT_ID` must be set? How about other variables such as `AZURE_CLIENT_ID`? Please update the document if needed.
https://aquasecurity.github.io/trivy/v0.41/docs/advanced/private-registries/acr/

Reply:
yes  all three env. variables must be set as describe in existing doc.
```
# must set TRIVY_USERNAME empty char
export AZURE_CLIENT_ID$(echo $SP_DATA | jq -r .appId)
export AZURE_CLIENT_SECRET$(echo $SP_DATA | jq -r .password)
export AZURE_TENANT_ID$(echo $SP_DATA | jq -r .tenant)
```

Reply:
How is it possible to use the workload identity within the trivy-operator helm chart?

Reply:
@KyrreHaugland follow this issue https://github.com/aquasecurity/trivy-operator/issues/1057 , I'll add support next week in operator